clients/src/main/java/org/apache/kafka/common/requests/JoinGroupRequest.java

```diff
@@ -60,14 +62,40 @@ public ByteBuffer metadata() {
         }
     }
 
+    // v0 constructor
+    @Deprecated
```

Comment:
Jun generally suggests this pattern when reviewing code. I personally don't like it because it clutters the build with warnings and often there's no plan to remove support for older protocol versions. I have tried to change his mind, but failed so far. :)